### DISCUSSION TITLE
How to load multiple view defination in one single viewport

### DISCUSSION THREAD

**Discussion by hsanand61564** [iTwin/itwinjs-core#181] [Q&A] [answered]
Hi All

I want to multiple multiple view definition on 

<SimpleViewport
          imodel={this.props.imodel}
          viewDefinitionId={this.props.viewDefinitionId}
        />

Can anyone suggest a way out for the same, I would be very thankful to you for the same.

**Comment by pmconne** [comment]
A Viewport can only have one ViewDefinition (ViewState) at one time. The ViewState defines the camera/frustum; it doesn't make sense to talk about a viewport with more than one camera or frustum.
If you mean you'd like to switch the Viewport to use a different ViewState, see [Viewport.changeView](https://github.com/imodeljs/imodeljs/blob/fbddeb5040dbc273539c86549039ef7fb928eb45/core/frontend/src/Viewport.ts#L1976).
If I'm not understanding the question, please clarify.

**Comment by hsanand61564** [comment]
@pmconne  I want to say that , how can I load more than one viewdefinition id along with the initial viewdefinition ID, or we can merge  multiple viewstates. 
something like this.viewport.addViewedModels(models) but dont know exactly how to implement the same.

**Comment by pmconne** [comment, marked as answer]
I think it would be up to you to define how to merge the view definitions. Which bits of the second view definition do you want to merge into the first view definition?

**Comment by hsanand61564** [comment]
@pmconne The main purpose is to show entire model , but all the sample is only showing one object/item

**Reply by pmconne** [reply]
Are you talking about [these samples](https://www.itwinjs.org/sample-showcase/)?
They allow you to select between different **iModels**. Each **iModel** consists of multiple **models**.
As you already mentioned, `viewport.addViewedModels()` allows you to add additional models to the viewport. If you wanted to add all of the iModel's spatial models to the viewport, you could query to obtain their Ids and use `addViewedModels()`.
Is that what you are trying to do?

**Comment by hsanand61564** [comment]
@pmconne I got your point, but the main point where I am struggling is getting the viewport. 

![image](https://user-images.githubusercontent.com/60655118/98284582-56ca8d00-1fc7-11eb-831d-7f4a9e3c4ae6.png)

const view = IModelApp.viewManager.selectedView; ---> This line is returning undefined to us.

Also after the below statement
IModelApp.viewManager.selectedView?.addViewedModels
Do we need to do something that would refresh the viewport so as to show all the models added to it

plz suggest

**Comment by hsanand61564** [comment]
@pmconne Any suggestions plz

**Reply by pmconne** [reply]
You have just started up `IModelApp`. As I mentioned, iModel.js does not open any `Viewport`s automatically; that's up to the application. So you cannot find a viewport until you create one.
Are you basing your application on an existing sample application? If so, consider studying the code a bit to understand how it works.
